src/dra/75F54.c

```diff
@@ -178,7 +178,74 @@ INCLUDE_ASM("asm/us/dra/nonmatchings/75F54", func_80116B0C);
 
 INCLUDE_ASM("asm/us/dra/nonmatchings/75F54", func_801177A0);
 
-INCLUDE_ASM("asm/us/dra/nonmatchings/75F54", func_80117AC0);
+void func_80117AC0(void) {
+    Collider sp10;
+    s32 collisionCount;
+    CheckCollision(PLAYER.posX.i.hi, PLAYER.posY.i.hi + 0x19, &sp10, 0);
+    collisionCount = (s32)sp10.unk0 & 1;
+    CheckCollision(PLAYER.posX.i.hi + 4, PLAYER.posY.i.hi + 0x19, &sp10, 0);
+    if ((s32)sp10.unk0 & 1) {
+        collisionCount += 1;
+    }
+    CheckCollision(PLAYER.posX.i.hi - 4, PLAYER.posY.i.hi + 0x19, &sp10, 0);
+    if ((s32)sp10.unk0 & 1) {
+        collisionCount += 1;
+    }
+    if ((g_Player.pl_vram_flag & 0x41) == 0x41) {
+        collisionCount += 1;
+    }
+    PLAYER.unk1E = 0;
+    func_8010E27C();
+    if (collisionCount == 0) {
+        func_8010E7AC();
+        return;
+    }
+    switch (PLAYER.step_s) {
+    case 0:
+        if (PLAYER.animFrameDuration < 0) {
+            func_8010DA48(0xF6);
+            PLAYER.step_s += 1;
+        }
+        return;
+    case 1:
+        // If any directional button is pressed
+        if (g_Player.padTapped & 0xF000) {
```

Comment:
```suggestion
if (g_Player.padTapped & (PAD_UP + PAD_RIGHT + PAD_DOWN + PAD_LEFT)) {

Reply:
How would you feel about PAD_UP | PAD_RIGHT | PAD_DOWN | PAD_LEFT rather than plusses?

Reply:
if it matches, you should go with the |, yeah